hikari/impl/event_manager_base.py

```diff
@@ -306,26 +302,41 @@ def filtered(
     else:
         event_types = event_types.dispatches()
 
+    bitmask = 0
+    for event_type in event_types:
+        bitmask |= event_type.bitmask()
+
+    # https://github.com/python/mypy/issues/2087
     def decorator(method: UnboundMethodT[EventManagerBaseT], /) -> UnboundMethodT[EventManagerBaseT]:
         method.__cache_components__ = cache_components  # type: ignore[attr-defined]
-        method.__event_types__ = event_types  # type: ignore[attr-defined]
+        method.__events_bitmask__ = bitmask  # type: ignore[attr-defined]
         assert isinstance(method, _FilteredMethodT), "Incorrect attribute(s) set for a filtered method"
         return method  # type: ignore[unreachable]
 
     return decorator
 
 
-@attr.define(hash=True)
+@attr.define(weakref_slot=False, hash=True)
```

Comment:
```suggestion
@attr.define(weakref_slot=False)
```
And then remove `hash=False` from attr.field below